src/GHC/Util.hs

```diff
@@ -83,13 +88,23 @@ baseDynFlags :: DynFlags
 baseDynFlags = foldl' xopt_set
              (defaultDynFlags fakeSettings fakeLlvmConfig) enabledExtensions
 
-parsePragmasIntoDynFlags ::
-  DynFlags -> FilePath -> String -> IO (Either String DynFlags)
-parsePragmasIntoDynFlags flags filepath str =
+-- | Adjust the input 'DynFlags' to take into account language
+-- extensions to explicitly enable/disable as well as language
+-- extensions enabled by pragma in the source.
+parsePragmasIntoDynFlags :: DynFlags
+                         -> ([Extension], [Extension])
+                         -> FilePath
+                         -> String
+                         -> IO (Either String DynFlags)
+parsePragmasIntoDynFlags flags (enable, disable) filepath str =
   catchErrors $ do
     let opts = getOptions flags (stringToStringBuffer str) filepath
     (flags, _, _) <- parseDynamicFilePragma flags opts
-    return $ Right flags
+    -- Explicit enable extensions
+    let flags' =  foldl' xopt_set flags enable
+    -- Explicit disable extensions
```

Comment:
I don't find this comment that valuable - it seems pretty obvious from the code

Reply:
Fair.